docs/guides/migration/10.0.0.md

```diff
@@ -0,0 +1,108 @@
+
+Use `contractDecodeCallResultAPI` instead.
+
+### removed `setOptions` on contract instance ([b88e767](https://github.com/aeternity/aepp-sdk-js/commit/b88e767c86874f259ff7a1eb1784c368524e7167))
+
+Pass them through `getContractInstance` options instead.
+
+### additional options of `getContractInstance` accepted as usual ones ([10fb7ba](https://github.com/aeternity/aepp-sdk-js/commit/10fb7bad8f61b973c1be5daec50909d251ec1f90))
+
+For example, replace
+```js
+sdk.getContractInstance(source, { opt: { ttl: 1 } })
+```
+with
+```js
+sdk.getContractInstance(source, { ttl: 1 })
+```
+
+### `contractCallStatic`, `contractCall`, `contractDeploy` are deprecated now ([c4ec019](https://github.com/aeternity/aepp-sdk-js/commit/c4ec019372f5d5745378e781afb1eb6b2b5acfeb))
+
+Use `getContractInstance` instead. Also, these methods will accept JavaScript-type variables instead of
+Sophia-encoded. For example:
+```js
+sdk.contractCallStatic(source, address, methodName, ['42'])
+```
+should be replaced with
+```js
+sdk.contractCallStatic(source, address, methodName, ['42'])
+```
+
+### dropped compatibility with compiler@4.1.0 ([f9cef12](https://github.com/aeternity/aepp-sdk-js/commit/f9cef12a7db4b2559519ef7f8380cc9e89630492))
+
+Use compiler 6.0.0 and above.
+
+### invert and rename `forceCodeCheck` option to `validateBytecode` in `getContractInstance` ([72122fa](https://github.com/aeternity/aepp-sdk-js/commit/72122facdcec9921202b210c387c19509b26e578))
+
+Use `validateBytecode` when you need to ensure that the source code/bytecode provided to
+`getContractInstance` corresponds to the on-chain bytecode.
+
+### removed `getConsensusProtocolVersion` method ([75f0447](https://github.com/aeternity/aepp-sdk-js/commit/75f044792bd2d81eca3a2abc8218e8ca5f167134))
+
+Use `node.consensusProtocolVersion` instead.
+
+### switched to @aeternity/aepp-calldata package ([#1313](https://github.com/aeternity/aepp-sdk-js/pull/1313))
+
+Numbers in Sophia are not limited in size. Before they were returned from contract methods as usual
+JavaScript numbers that have limited accuracy. To fix this, we make it return instances of BigInt.
+
+Variant types are now supported by sdk, so replace `"RelativeTTL(50)"` with `{ RelativeTTL: [50] }`.
+As an exception `Some(value)` is converted to the exact value in JavaScript and `None` is converted to `undefined`
+(Sophia's `option` type).
+
+Contract methods will accept/return instances of JavaScript's `Map` as variables of Sophia's `map`
+type.
+
+Sophia's `hash`, `signature`, `bytes` types return values as `Uint8Array` instead of a hex-encoded
+string.
+
+Check the [documentation](https://github.com/aeternity/aepp-calldata-js/blob/master/README.md)
+of calldata package for additional info.
+
+Use `.decodedResult` instead of `.decode()` to get the result of method call.
+
+### pass source in options of `sdk.getContractInstance` ([5c690d2](https://github.com/aeternity/aepp-sdk-js/commit/5c690d2d77a433c8496ab9cda6fd66e52a6b4b23))
```

Comment:
maybe we can also add an example to pass aci?

Reply:
I intended to write the exactly migration guide, without mentioning new features, but it is ok to add this

Reply:
also fine by me